src/components/manager/runs/RunLastContent.js

```diff
@@ -16,47 +17,111 @@ const COLUMNS_LAST_CONTENT = [
+        : fileNames;
+
+    const divStyle = isTruncated
+        ? {
+            display: "flex",
+            flexDirection: "column",
+            alignItems: "flex-start",
+            justifyContent: "center",
+            cursor: "pointer",
+        }
+        : {};
+
+    const openModal = () => isTruncated && setIsModalVisible(true);
+
+    const closeModal = () => setIsModalVisible(false);
+
+    return (
+        <>
+            <div onClick={openModal} style={divStyle}>
+                {truncatedFileNames.map((element, index) =>
+                    typeof element === "string" ? <div key={index}>{element}</div> : element,
+                )}
+            </div>
+            <Modal
+                title={`${dataType.charAt(0).toUpperCase() + dataType.slice(1)} files`}
+                footer={null}
+                visible={isModalVisible}
+                onCancel={closeModal}
+                bodyStyle={{maxHeight: "80vh", overflowY: "auto", whiteSpace: "text-overflow"}}
+            >
+                 <ul style={{ padding: "0 20px", listStyle: "none" }}>
+                    {fileNames.map((fileName, index) => (
+                        <li key={index} style={modalListStyle}>- {fileName}</li>
+                    ))}
+                </ul>
+            </Modal>
+        </>
+    );
+}
+
+FileNamesCell.propTypes = {
+    fileNames: PropTypes.arrayOf(PropTypes.string).isRequired,
+    dataType: PropTypes.string.isRequired,
+};
+
 const buildKeyFromRecord = (record) => `${record.dataType}-${record.tableId}`;
 
+const fileNameFromPath = (path) => path.split("/").at(-1);
+
+const getFileInputsFromWorkflowMetadata = (workflowMetadata) => {
+    return workflowMetadata.inputs
+        .map(input => `${workflowMetadata.id}.${input.id}`);
+};
+
+const getDateFromEndTime = (endTime) => endTime.split("T")[0];
+
 const processIngestions = (data, currentTables) => {
-    const currentTableIds = (currentTables || []).map((table) => table.table_id);
+    const currentTableIds = new Set((currentTables || []).map((table) => table.table_id));
 
-    const ingestionsByDataType = data.reduce((acc, obj) => {
-        if (obj.state === "COMPLETE") {
-            const dataType = obj.details.request.tags.workflow_metadata.data_type;
-            const tableId = obj.details.request.tags.table_id;
+    const ingestionsByDataType = data.reduce((ingestions, run) => {
+        if (run.state === "COMPLETE") {
+            const dataType = run.details.request.tags.workflow_metadata.data_type;
+            const tableId = run.details.request.tags.table_id;
 
-            if (tableId === undefined || !currentTableIds.includes(tableId)) {
-                return acc;
+            if (tableId === undefined || !currentTableIds.has(tableId)) {
+                return ingestions;
             }
+            const fileNameKey = getFileInputsFromWorkflowMetadata(run.details.request.tags.workflow_metadata);
```

Comment:
some workflow types can have more than one file input - so there may be multiple keys.